### DISCUSSION TITLE
sdist is missing tests

### DISCUSSION THREAD

**Discussion by mtelka** [encode/httpcore#796] [Potential Issue] [answered]
The sdist package version 0.18.0 at PyPI is missing tests.  This is regression from version 0.17.3.  Would it be possible to add tests back to make downstream testing easier?

Thank you.

**Comment by karpetrosyan** [comment]
We are following the structure of httpx by not including tests in the sdist. Could you please tell me when it will be useful? Is it worthwhile to increase the package size by including project tests?

**Comment by mtelka** [comment]
I'm packaging `httpcore` for OpenIndiana and we use sdist as our primary source for packages.  During packaging we run tests to make sure everything works as expected.  Actually, most Python projects (more than 90% of those packaged for OpenIndiana) include tests in sdist.

Re the package size: The sdist size for httpcore 0.17.3 at PyPI is 62.7 kB (this contained tests), while the sdist size for httpcore 0.18.0 is 60.4 kB (without tests).  I doubt anybody would notice the sdist size difference :-).

Thank you.

**Reply by lovelydinosaur** [reply]
> more than 90% of those packaged for OpenIndiana

Could you point me towards what a few popular packages do here?

Three that come to mind...

* flask
* requests
* django

What is their tests/packaging situation?

**Comment by karpetrosyan** [comment, marked as answer]
Thank you so much! I assumed the size difference was greater than 2 kB :smile: . 
Would you like to [create an issue](https://github.com/encode/httpcore/issues/new?assignees=&labels=&projects=&template=1-issue.md) for this?

**Reply by mtelka** [reply]
Done: #798

**Comment by mtelka** [comment]
> > more than 90% of those packaged for OpenIndiana
> 
> Could you point me towards what a few popular packages do here?

Here is list of Python projects packaged for OpenIndiana (550+ projects): https://github.com/OpenIndiana/oi-userland/tree/oi/hipster/components/python

> 
> Three that come to mind...
> 
>     * flask
> 
>     * requests
> 
>     * django
> 
> 
> What is their tests/packaging situation?

Both flask and requests does have tests in sdist.  For django I'm not sure because we do not have it packaged for OpenIndiana (yet).
